contrib/babelfishpg_common/src/spatialtypes.c

```diff
@@ -0,0 +1,134 @@
+    GSERIALIZED *geom; /* Used to Store the bytes in the Format which is stored in PostGIS */
+
+    /* Ensure LWGEOM_in_p is properly initialized before using it */
+    if (LWGEOM_in_p == NULL) LWGEOM_in_p = (LWGEOM_in_t) load_external_function("$libdir/postgis-3", "LWGEOM_in", true, NULL);
+
+    /* Call the LWGEOM_in function via the function pointer */
+    geom_datum = LWGEOM_in_p(fcinfo);
+
+    geom = (GSERIALIZED*)PG_DETOAST_DATUM(geom_datum);
+
+    /* check if it is a 2-D point type */
+    if(*((uint32_t*)geom->data) != 1) ereport(ERROR,
+                                        (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+                                        errmsg("Currently only Point type is supported")));
+    else
+    {
+        if ( (geom->gflags & 0x01) || (geom->gflags & 0x02) || (geom->gflags & 0x03) )
+        {
+            ereport(ERROR,
+                    (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+                    errmsg("Unsupported flags")));
+        }
+        PG_RETURN_DATUM(geom_datum);
+    }
+    
+    PG_RETURN_DATUM(geom_datum);
+
+}
+
+Datum
+geography_in(PG_FUNCTION_ARGS)
+{
+    Datum geom_datum;
+    GSERIALIZED *geom; /* Used to Store the bytes in the Format which is stored in PostGIS */
+    char *input_str = PG_GETARG_CSTRING(0);
+    float8 lat;
+    bool isBinary = false;
+
+    if(input_str[0] == '0') isBinary = true;
+
+    /* Ensure LWGEOM_in_p is properly initialized before using it */
+    if (LWGEOM_in_p == NULL) LWGEOM_in_p = (LWGEOM_in_t) load_external_function("$libdir/postgis-3", "LWGEOM_in", true, NULL);
+
+    /* Call the LWGEOM_in function via the function pointer */
+    geom_datum = LWGEOM_in_p(fcinfo);
+
+    geom = (GSERIALIZED*)PG_DETOAST_DATUM(geom_datum);
+
+    if (gserialized_set_srid_p == NULL) gserialized_set_srid_p = (gserialized_set_srid_t) load_external_function("$libdir/postgis-3", "LWGEOM_set_srid", true, NULL);
+    
+    fcinfo->args[0].value = geom_datum;
+    fcinfo->args[1].value = Int32GetDatum(4326);
+    
+    /* Setting deafault SRID for geography datatype = 4326 */
+    geom_datum = gserialized_set_srid_p(fcinfo);
+
+    /* check if it is a 2-D point type */
+    if(*((uint32_t*)geom->data) != 1) ereport(ERROR,
+                                        (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+                                        errmsg("Currently only Point type is supported")));
+    else
+    {
+        if ((geom->gflags & 0x01) || (geom->gflags & 0x02) || (geom->gflags & 0x03) )
+            ereport(ERROR,
+                    (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+                    errmsg("Unsupported flags")));
+        else
+        {
+            if (ST_FLIP_COORD_p == NULL) ST_FLIP_COORD_p = (ST_FLIP_COORD_t) load_external_function("$libdir/postgis-3", "ST_FlipCoordinates", true, NULL);
+            if (lwgeom_x_p == NULL) lwgeom_x_p = (lwgeom_x_t) load_external_function("$libdir/postgis-3", "LWGEOM_x_point", true, NULL);
```

Comment:
Nit: Keep if condition and statement in separate lines.

Reply:
Done